ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts

```diff
@@ -0,0 +1,105 @@
+import { Either, left, right } from "fp-ts/lib/Either";
+import { NavigationActions } from "react-navigation";
+import { SagaIterator } from "redux-saga";
+import { call, Effect, put, race, select, take } from "redux-saga/effects";
+import { ActionType, getType } from "typesafe-actions";
+import { navigateToWalletHome } from "../../../../../../store/actions/navigation";
+import { navigationHistoryPop } from "../../../../../../store/actions/navigationHistory";
+import { navigationCurrentRouteSelector } from "../../../../../../store/reducers/navigation";
+import { SagaCallReturnType } from "../../../../../../types/utils";
+import { isReady } from "../../../model/RemoteValue";
+import {
+  navigateToBpdOnboardingDeclaration,
+  navigateToBpdOnboardingInformationTos,
+  navigateToBpdOnboardingLoadActivationStatus
+} from "../../../navigation/action";
+import BPD_ROUTES from "../../../navigation/routes";
+import { bpdLoadActivationStatus } from "../../../store/actions/details";
+import {
+  bpdOnboardingAcceptDeclaration,
+  bpdOnboardingCancel,
+  bpdUserActivate
+} from "../../../store/actions/onboarding";
+import { bpdEnabledSelector } from "../../../store/reducers/details/activation";
+
+export const isLoadingScreen = (screenName: string) =>
+  screenName === BPD_ROUTES.ONBOARDING.LOAD_CHECK_ACTIVATION_STATUS;
+
+export function* isBpdEnabled(): Generator<
+  Effect,
+  Either<Error, boolean>,
+  any
+> {
+  const remoteActive: ReturnType<typeof bpdEnabledSelector> = yield select(
+    bpdEnabledSelector
+  );
+  if (isReady(remoteActive)) {
+    return right<Error, boolean>(remoteActive.value);
+  } else {
+    yield put(bpdLoadActivationStatus.request());
+    const bpdActivationResults: ActionType<
+      | typeof bpdLoadActivationStatus.success
+      | typeof bpdLoadActivationStatus.failure
+    > = yield take([
+      getType(bpdLoadActivationStatus.success),
+      getType(bpdLoadActivationStatus.failure)
+    ]);
+    return bpdActivationResults.type ===
+      getType(bpdLoadActivationStatus.success)
+      ? right<Error, boolean>(bpdActivationResults.payload.enabled)
+      : left<Error, boolean>(bpdActivationResults.payload);
```

Comment:
same of above to avoid some boiler-plate

```typescript
const bdpEnabled = yield call(() => getAsyncResult(bpdLoadActivationStatus,undefined as void, cit => cit.enabled));
```

Reply:
Like before, I need to distinguish between the failure (Error) and the success with the result (true,false) because I've two different flow (retry or continue the flow), for this reason i need an`Either<Error,boolean>`

Reply:
I've cherry picked your commit and refactored the functions with the new reusable code.